src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,54 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact in the address book.
+ */
+public class Contact {
+    
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given address.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */
+    public Contact(String contact, boolean isPrivate, String messageConstraints, String validRegex) throws IllegalValueException {
+        String trimmedContact = contact.trim();
+        this.isPrivate = isPrivate;
+        if (!isValid(trimmedContact, validRegex)) {
+            throw new IllegalValueException(messageConstraints);
+        }
+        this.value = trimmedContact;
+    }
+
+    /**
+     * Returns true if a given string is a valid contact.
+     */
+    public static boolean isValid(String test, String validRegex) {
+        return test.matches(validRegex);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
This is fine, but note that now an Email and Address having the same value will be considered equal even if they are of different types.

If you decide to override the equals() method then hashCode() has to be overridden